cdc/owner.go

```diff
@@ -343,7 +338,11 @@ func (o *Owner) newChangeFeed(
 		}
 	}()
 	go func() {
-		err := <-errCh
+		var err error
+		select {
+		case <-ctx.Done():
+		case err = <-errCh:
+		}
 		if errors.Cause(err) != context.Canceled {
 			log.Error("error on running owner", zap.Error(err))
 		} else {
```

Comment:
If the branch `<-ctx.Done()` is fired, `err` is nil in L347, the error log is not as wished.

Reply:
Yes, but I haven't any better solution. And when the ctx is done, the error from the sink must be canceled.

Reply:
I mean if `ctx.Done` is triggered first, the errCh is not fetched and `err` is nil, we should log as normal exit then.

Reply:
updated